### PR TITLE
General stuff

### PR DESCRIPTION
**Pull request by MissAllSunday**
Just some general stuff:
- Fixed some typos.
- Added a description to loadGeneralSettingParameters().
- Add a function to load custom profile fields without the need to use loadMemberContext() it adds the values to $memberContext but it was designed to be used as a stand alone function to use when you only need to use the profile fields.

### CODE REVIEW

**Comment by MissAllSunday** [comment]
OK, for the loadMemberCustomFields($users) function makes any sense it gotta have an extra param, a title param to get specific rows and not the entire themes table... or better yet, an array or titles to be able to get multiple profile fields at once.

I'll get to that later, kudos goes to Oldiesmann for letting me know about this on IRC.

**Comment by MissAllSunday** [comment]
OK https://github.com/MissAllSunday/SMF2.1/commit/e2bf6e03ea178fb763cd5f69883f888af49eb8d0 sets a better and cleaner version. It nows pass all possible variables for each field and supports multiple field names as well as multiple users.

File: Sources/Load.php [hunk 1]
```diff
@@ -1243,6 +1243,72 @@ function loadMemberContext($user, $display_custom_fields = false)
...
+	$params = !is_array($params) ? array($params) : array_unique($params);
+	$return = array();
+
+	$request = $smcFunc['db_query']('', '
+		SELECT c.id_field, c.col_name, c.field_name, c.field_desc, c.field_type, c.field_length, c.field_options, c.mask, show_reg,
+		c.show_display, c.show_profile, c.private, c.active, c.bbc, c.can_search, c.default_value, c.enclose, c.placement, t.variable, t.value, t.id_member
+		FROM {db_prefix}themes AS t
+			LEFT JOIN {db_prefix}custom_fields AS c ON (c.col_name = t.variable)
+		WHERE id_member' . (count($users) == 1 ? ' = {int:loaded_ids} ' : ' IN ({array_int:loaded_ids}) ') .
+			'AND variable' . (count($params) == 1 ? ' = {string:params}' : ' IN ({array_string:params})'),
```
Comment (emanuele45) [inline Sources/Load.php:1271]: Why not use just IN?
That would save you some comparisons:

```
    WHERE id_member IN ({array_int:loaded_ids})
      AND variable IN ({array_string:params})',
```

and here below:

```
      'loaded_ids' => $users,
      'params' => $params,
```

Comment (MissAllSunday) [inline Sources/Load.php:1271]: Because thats how SMF do things... I just followed and copy/paste SMF coding style.

Comment (emanuele45) [inline Sources/Load.php:1271]: Okay...never checked before... :angel: 

/me feels is a bit odd
I'm not sure if there is any practical difference...

Comment (MissAllSunday) [inline Sources/Load.php:1271]: There isn't if you aren't picky in terms of performance, it is if you want the code to be more organized but there is a bazillion other places where SMF does this, might as well go with the flow.

Comment (norv) [inline Sources/Load.php:1271]: SMF avoids using 'IN' in the query, for the query to use the index instead. There's only an index on id_member iirc however.

File: Sources/Load.php [hunk 2]
```diff
@@ -1243,6 +1243,72 @@ function loadMemberContext($user, $display_custom_fields = false)
...
+		array(
+			'loaded_ids' => (int) count($users) == 1 ? $users[0] : $users,
+			'params' => (string) count($params) == 1 ? $params[0] : $params,
+		)
```
Comment (norv) [inline Sources/Load.php:1275]: I'm not sure what is the intention of these casts, but they won't work as (probably) expected

Comment (MissAllSunday) [inline Sources/Load.php:1275]: .___. again, I just copy/paste SMF code... you should point that out to whoever wrote that ten years ago.

Comment (norv) [inline Sources/Load.php:1275]: Could you please point me to an example?

File: Sources/Load.php (hunk 1 above)
Comment (emanuele45) [inline Sources/Load.php:1271]: From my (quick) test with EXPLAIN, MySQL is smart enough to understand there is just one value and uses the index anyway. Of course if there is more than one then it doesn't.

Comment (norv) [inline Sources/Load.php:1271]: Cool, thanks for checking it out!

**Pull request closed without merging**